crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -189,53 +105,51 @@ impl AuthorityStorePruner {
+                batch_effects = vec![];
+                checkpoints_in_batch = 0;
+            }
+        }
+        if !batch_effects.is_empty() {
+            Self::handle_checkpoint(batch_effects, &perpetual_db.objects)?;
+            checkpoint_store
+                .update_highest_pruned_checkpoint(checkpoint_number, checkpoint_digest)?;
         }
+        debug!(
+            "Finished pruner iteration. Latest pruned checkpoint: {}",
+            checkpoint_number
+        );
+        Ok(())
     }
 
     fn setup_objects_pruning(
-        num_versions_to_retain: u64,
-        pruning_timeperiod: Duration,
-        pruning_initial_delay: Duration,
         num_epochs_to_retain: u64,
         epoch_duration_ms: u64,
         perpetual_db: Arc<AuthorityPerpetualTables>,
         checkpoint_store: Arc<CheckpointStore>,
     ) -> Sender<()> {
         let (sender, mut recv) = tokio::sync::oneshot::channel();
-        info!(
-            "Starting object pruning service with num_versions_to_retain={num_versions_to_retain}"
-        );
-        let mut prune_interval =
-            tokio::time::interval_at(Instant::now() + pruning_initial_delay, pruning_timeperiod);
-        prune_interval.set_missed_tick_behavior(time::MissedTickBehavior::Skip);
-
+        debug!("Starting object pruning service with num_epochs_to_retain={num_epochs_to_retain}");
         let duration_ms = if num_epochs_to_retain > 0 {
             epoch_duration_ms / 2
         } else {
             1000
         };
-        let mut live_prune_interval = tokio::time::interval(Duration::from_millis(duration_ms));
+        let mut prune_interval = tokio::time::interval(Duration::from_millis(duration_ms));
 
         tokio::task::spawn(async move {
             loop {
                 tokio::select! {
-                    _ = prune_interval.tick(), if num_versions_to_retain != u64::MAX => {
-                        info!("Starting pruning of objects table");
-                        let num_pruned = Self::prune_objects(num_versions_to_retain, &perpetual_db.objects);
-                        info!("Finished pruning with total object versions pruned = {}", num_pruned);
-                        if let Ok(()) = perpetual_db.objects.flush() {
-                            info!("Completed flushing objects table");
-                        } else {
-                            error!("Failed to flush objects table");
-                        }
-                    },
-                    _ = live_prune_interval.tick(), if num_epochs_to_retain != u64::MAX => {
-                        match Self::process_checkpoints(&perpetual_db, &checkpoint_store, num_epochs_to_retain) {
-                            Ok(()) => info!("Pruned checkpoints"),
-                            Err(err) => error!("Failed to prune objects: {:?}", err),
+                    _ = prune_interval.tick(), if num_epochs_to_retain != u64::MAX => {
```

Comment:
I think this interval should be in hours not milliseconds, since my understanding of pruning is that it deletes past epochs, which are unlikely to have changed every few millis.

Reply:
same, related to `num_epochs_to_retain = 0`